packages/flutter/lib/src/painting/image_stream.dart

```diff
@@ -677,6 +677,14 @@ class MultiFrameImageStreamCompleter extends ImageStreamCompleter {
 
   Future<void> _decodeNextFrameAndSchedule() async {
     try {
+      if (!SchedulerBinding.instance.framesEnabled) {
+        SchedulerBinding.instance.scheduleFrameCallback((_) {
+          if (hasListeners && _codec != null) {
+            _decodeNextFrameAndSchedule();
+          }
+        });
+        return;
+      }
```

Comment:
This can potentially be racy, e.g the app may go to the background just after we checked `framesEnabled`.